test/e2e-tests.sh

```diff
@@ -28,42 +28,99 @@ source $(dirname $0)/../vendor/github.com/knative/test-infra/scripts/e2e-tests.s
   wait_until_pods_running knative-eventing || fail_test "Failed to install the In-Memory ClusterChannelProvisioner"
+
+  echo "Installing GCPPubSub ClusterChannelProvisioner"
+  gcppubsub_setup
+  # TODO(Fredy-Z): delete this flag after https://github.com/knative/test-infra/pull/692 is merged and updated
+  E2E_PROJECT_ID="$(gcloud config get-value project)"
+  sed "s/REPLACE_WITH_GCP_PROJECT/${E2E_PROJECT_ID}/" ${GCP_PUBSUB_PROVISIONER_CONFIG} > ${GCP_PUBSUB_TEMP_CONFIG}
+  ko apply -f ${GCP_PUBSUB_TEMP_CONFIG}
+  wait_until_pods_running knative-eventing || fail_test "Failed to install the GCPPubSub ClusterChannelProvisioner"
 }
 
+# Teardown the Knative environment after tests finish.
 function knative_teardown() {
   echo ">> Stopping Knative Eventing"
   echo "Uninstalling Knative Eventing"
-  ko delete --ignore-not-found=true --now --timeout 60s -f config/
+  ko delete --ignore-not-found=true --now --timeout 60s -f ${EVENTING_CONFIG}
 
-  wait_until_object_does_not_exist namespaces knative-eventing
+  echo "Uninstalling In-Memory ClusterChannelProvisioner"
+  ko delete --ignore-not-found=true --now --timeout 60s -f ${IN_MEMORY_CHANNEL_PROVISIONER_CONFIG}
+
+  echo "Uninstalling GCPPubSub ClusterChannelProvisioner"
+  gcppubsub_teardown
+  ko delete --ignore-not-found=true --now --timeout 60s -f ${GCP_PUBSUB_TEMP_CONFIG}
 
-  wait_until_object_does_not_exist customresourcedefinitions subscriptions.eventing.knative.dev
-  wait_until_object_does_not_exist customresourcedefinitions channels.eventing.knative.dev
+  wait_until_object_does_not_exist namespaces knative-eventing
 }
 
-# Setup resources common to all eventing tests
+# Setup resources common to all eventing tests.
 function test_setup() {
-  # Publish test images
+  # Publish test images.
   echo ">> Publishing test images"
   $(dirname $0)/upload-test-images.sh e2e || fail_test "Error uploading test images"
 }
 
+readonly PUBSUB_SERVICE_ACCOUNT="eventing_pubsub_test"
+readonly PROJECT_ID="${GCP_PROJECT}"
+readonly PUBSUB_SERVICE_ACCOUNT_KEY="knative-gcppubsub-channel.json"
+
+# Create resources required for GCP PubSub provisioner setup
+function gcppubsub_setup() {
+  local service_account_key=${GOOGLE_APPLICATION_CREDENTIALS}
+  if (( ! IS_PROW )); then
+    echo "Set up ServiceAccount for GCP PubSub provisioner"
+    gcloud services enable pubsub.googleapis.com
+    gcloud iam service-accounts create ${PUBSUB_SERVICE_ACCOUNT}
+    gcloud projects add-iam-policy-binding ${PROJECT_ID} \
+    --member=serviceAccount:${PUBSUB_SERVICE_ACCOUNT}@${PROJECT_ID}.iam.gserviceaccount.com \
+    --role roles/pubsub.editor
+    gcloud iam service-accounts keys create ${PUBSUB_SERVICE_ACCOUNT_KEY} \
+    --iam-account=${PUBSUB_SERVICE_ACCOUNT}@${PROJECT_ID}.iam.gserviceaccount.com
+    service_account_key="${PUBSUB_SERVICE_ACCOUNT_KEY}"
+  fi
+
+  kubectl -n knative-eventing create secret generic gcppubsub-channel-key --from-file=key.json=${service_account_key}
```

Comment:
`gcppubsub-channel-key` should be a constant.

Reply:
Done.